scan_rr.go

```diff
@@ -133,7 +133,8 @@ func (rr *A) parse(c *zlexer, o, f string) *ParseError {
 	}
 
 	rr.A = net.ParseIP(l.token)
-	if rr.A == nil || l.err {
+	isIPv6 := strings.Contains(l.token, ":")
```

Comment:
A better way (and use in other places in this lib) is to use rr.A.To4() and check if that's `nil` or not

Reply:
I considered that, but that causes test failures because `net.ParseIP("::FFFF:127.0.0.1").To4()` is non-nil.

Reply:
https://github.com/asaskevich/govalidator/pull/100 is another project having observed the same issue.

Reply:
https://play.golang.org/p/zIBXCABhfax
seems checking for colons is the only way :(

Reply:
please add a comment saying this

Reply:
Done.